appservice/src/intTest/groovy/nu/yona/server/VpnConnectionStatusEventTest.groovy

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Stichting Yona Foundation
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v.2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server
+
+import static nu.yona.server.test.CommonAssertions.*
+
+import groovy.json.*
+import nu.yona.server.test.User
+
+class VpnConnectionStatusEventTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Richard posts initial VPN connected event: connection status updated and buddies informed'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+
+		when:
+		def response = richard.devices[0].postVpnStatus(appService, true)
+
+		then:
+		assertResponseStatusNoContent(response)
+
+		User richardAfterReload = appService.reloadUser(richard)
+		User bobAfterReload = appService.reloadUser(bob)
+		!richard.devices[0].vpnConnected
```

Comment:
We could move this to a simple test "Initially when Richard is created, VPN status is disconnected" and leave it out of these tests.

Reply:
The main purpose of these assertions is not to guard the initial status but to guard the test itself. Does the test indeed see a change from "disconnected" to "connected"?